compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1343,6 +1343,15 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
         }
     }
 
+    fn encode_attributes(&mut self, def_id: DefId) {
+        let attrs = self
+            .tcx
+            .get_attrs(def_id)
+            .iter()
+            .filter(|attr| !rustc_feature::is_builtin_only_local(attr.name_or_empty()));
```

Comment:
Does this keep non-builtin attributes like registered with `#![register_tool]`?

Reply:
yes, it only removes `builtin` attributes where `only_local` is true.

Reply:
:+1: